packages/phishing-controller/src/PhishingController.test.ts

```diff
@@ -25,213 +26,393 @@ describe('PhishingController', () => {
+      });
+      await controller.updateStalelist();
 
-      expect(controller.isOutOfDate()).toBe(false);
+      expect(controller.isStalelistOutOfDate()).toBe(false);
     });
 
     it('should not be out of date if the phishing lists were recently updated', async () => {
       const clock = sinon.useFakeTimers();
-      const controller = new PhishingController({ refreshInterval: 10 });
-      await controller.updatePhishingLists();
-      await clock.tick(5);
+      const controller = new PhishingController({
+        stalelistRefreshInterval: 10,
+      });
+      await controller.updateStalelist();
+      await clock.tick(1000 * 5);
 
-      expect(controller.isOutOfDate()).toBe(false);
+      expect(controller.isStalelistOutOfDate()).toBe(false);
     });
 
     it('should be out of date if the time elapsed since the last update equals the refresh interval', async () => {
       const clock = sinon.useFakeTimers();
-      const controller = new PhishingController({ refreshInterval: 10 });
-      await controller.updatePhishingLists();
-      clock.tick(10);
+      const controller = new PhishingController({
+        stalelistRefreshInterval: 10,
+      });
+      await controller.updateStalelist();
+      clock.tick(1000 * 10);
 
-      expect(controller.isOutOfDate()).toBe(true);
+      expect(controller.isStalelistOutOfDate()).toBe(true);
     });
   });
 
-  it('should be able to change the refreshInterval', async () => {
+  describe('isHotlistOutOfDate', () => {
+    it('should not be out of date upon construction', () => {
+      sinon.useFakeTimers();
+      const controller = new PhishingController({
+        hotlistRefreshInterval: 10,
+      });
+
+      expect(controller.isHotlistOutOfDate()).toBe(false);
+    });
+
+    it('should not be out of date after some of the refresh interval has passed', () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({
+        hotlistRefreshInterval: 10,
+      });
+      clock.tick(1000 * 5);
+
+      expect(controller.isHotlistOutOfDate()).toBe(false);
+    });
+
+    it('should be out of date after the refresh interval has passed', () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({
+        hotlistRefreshInterval: 10,
+      });
+      clock.tick(1000 * 10);
+
+      expect(controller.isHotlistOutOfDate()).toBe(true);
+    });
+
+    it('should be out of date if the refresh interval has passed and an update is in progress', async () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({
+        hotlistRefreshInterval: 10,
+      });
+      clock.tick(1000 * 10);
+      const pendingUpdate = controller.updateHotlist();
+
+      expect(controller.isHotlistOutOfDate()).toBe(true);
+
+      // Cleanup pending operations
+      await pendingUpdate;
+    });
+
+    it('should not re-request when an update is in progress', async () => {
```

Comment:
Hmmm.... this test seems more appropriate for the `updateHotlist` block, rather than here in the `isHotlistOutOfDate` block. It does _use_ that method, but the behavior it's testing (that updates while one us pending won't duplicate the request is within `updateHotlist`.

Reply:
Oh I see, there is no `updateHotlist` describe block. This test module is a bit of a mess right now, my apologies. Some `describe` grouping had been introduced but we haven't yet grouped all of these by method yet.

Putting this test in the root describe block would be perfectly fine as well.